### ISSUE TITLE
Entity encoding issues

### ISSUE THREAD

**Issue by mjmasn** [SoapBox/jQuery-linkify#58]
I have a field with the value something like ```&amp;&```. We encode the entities before displaying so we get ```&amp;amp;&amp;``` However after linkify()-ing the element we get ```&&``` displayed, which is incorrect.

I guess the intent is to normalise entities like ```&``` and ```&amp;``` to plain ```&``` before parsing the contents for urls but it does give an undesired effect to non-url text.

The simplest case to test in the console would be:
```javascript
$('<div>&amp;amp;&</div>').linkify()
```
which returns:
```javascript
[<div>&&</div>]
```
Ideally this should return the untouched:
```javascript
[<div>&amp;amp;&</div>]
```

**Comment by nfrasser** [comment]
Can you confirm that it's not jQuery that's automatically parsing the HTML? Here's what I get when I ran this in the console (with and without Linkify)

![screen shot 2015-02-06 at 09 29 21](https://cloud.githubusercontent.com/assets/1693461/6080834/b47be97e-ade2-11e4-80d3-31bc45bc2a7e.png)

**Comment by mjmasn** [comment]
@nfrasser thanks for the quick reply, yeah I've just worked out it's being caused by jQuery. I think I get slightly different output though:

```javascript
$('<div>&amp;amp;&') // Note the second 'amp;' doesn't have a preceeding '&'
[ <div>&amp;&</div> ]

$('<div>&amp;amp;&').linkify()
[ <div>&&</div> ]
```

Have managed to fix, at least temporarily, by using jQuery's ```.text()``` and reordering some nl2br conversion code.

Thanks for the help :+1:

**Comment by nfrasser** [comment]
Ahh, I think I see the issue. That's pretty strange. HTML parsing issues like this will be fixed in 2.0 ( #51 )
